src/miniscript/context.rs

```diff
@@ -221,6 +223,39 @@ impl Any {
             transmute::<&Miniscript<Pk, Segwitv0>, &Miniscript<Pk, Any>>(ms)
         }
     }
+
+    pub(crate) fn from_smv_legacy(
+        smv: &SortedMultiVec<bitcoin::PublicKey, Legacy>,
+    ) -> &Miniscript<bitcoin::PublicKey, Any> {
+        let ms = &Miniscript::from_ast(smv.sorted_node()).unwrap();
+        // The fields Miniscript<Pk, Legacy> and Miniscript<Any, Legacy> only
+        // differ in PhantomData. This unsafe assumes that the unlerlying byte
+        // representation of the both should be the same. There is a Miri test
+        // checking the same.
+        unsafe {
+            use std::mem::transmute;
+            transmute::<&Miniscript<bitcoin::PublicKey, Legacy>, &Miniscript<bitcoin::PublicKey, Any>>(
+                ms,
+            )
+        }
```

Comment:
Can you add a miri test which detects the fact that you are returning a pointer to dead memory?

Also, you are returning a pointer to dead memory :P.

Reply:
The current implementation that creates a new miniscript returns a reference to a value from the function which is unsafe.  The correct way would be have additional information in the iterator about this. 

I think I can add this as a follow up PR as it more related to `SatisfiedConstraints` than the sortedmulti part. If you are okay, we can have these are unimplemented!() for now?

Reply:
To be clear - this is UB and we need to come up with a different solution

Reply:
Yeah I'm ok with just making them `unimplemented!()` for now. cc @justinmoon

Reply:
I left this `unimplemented!()`